Xamarin.Forms.Platform.iOS/Cells/CellTableViewCell.cs

```diff
@@ -95,5 +95,21 @@ internal static UITableViewCell GetNativeCell(UITableView tableView, Cell cell,
 
 			return nativeCell;
 		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (_disposed)
+				return;
+
+			if (disposing)
+			{
+				PropertyChanged = null;
+				_cell = null;
+			}
+
+			_disposed = true;
+
+			base.Dispose(disposing);
+		}
```

Comment:
One of the issues seems to be that `CellTableViewCell` is hanging onto `_cell`. This should release the reference.